components/server/src/ide-service.ts

```diff
@@ -84,18 +84,28 @@ export class IDEService {
         return newIDESettings;
     }
 
-    async resolveWorkspaceConfig(workspace: Workspace, user: User): Promise<ResolveWorkspaceConfigResponse> {
+    async resolveWorkspaceConfig(
+        workspace: Workspace,
+        user: User,
+        userSelectedIdeSettings?: IDESettings,
+    ): Promise<ResolveWorkspaceConfigResponse> {
         const use = await this.configCatClientFactory().getValueAsync("use_IDEService_ResolveWorkspaceConfig", false, {
             user,
         });
         if (use) {
-            return this.doResolveWorkspaceConfig(workspace, user);
+            return this.doResolveWorkspaceConfig(
+                workspace,
+                userSelectedIdeSettings || user.additionalData?.ideSettings,
```

Comment:
Without knowing the type of IDESettings: Instead of the top-level `||`, could it make sense to do a deep-(semantic?)merge?

Reply:
IDESetting is describing exactly one IDE preference, so I don't think this can or should be merged